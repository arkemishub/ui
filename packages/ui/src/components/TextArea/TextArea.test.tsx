/**
 * Copyright 2023 Arkemis S.r.l.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { TextArea } from "./index";
import { render } from "@testing-library/react";
import React from "react";

describe("Switch", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<TextArea />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render textarea", () => {
    const { getByTestId } = render(<TextArea />);
    expect(getByTestId("arke-textarea")).toBeInTheDocument();
  });

  test("should render placeholder", () => {
    const { getByTestId } = render(<TextArea placeholder="Write here" />);
    expect(getByTestId("arke-textarea")).toBeInTheDocument();
  });

  test("should render label", () => {
    const { getByTestId } = render(<TextArea label="Label" />);
    expect(getByTestId("arke-textarea-label")).toBeInTheDocument();
  });

  test("should render rows", () => {
    const { getByTestId } = render(<TextArea rows={5} />);
    expect(getByTestId("arke-textarea")).toBeInTheDocument();
  });

  test("should render cols", () => {
    const { getByTestId } = render(<TextArea cols={40} />);
    expect(getByTestId("arke-textarea")).toBeInTheDocument();
  });

  test("should render disabled textarea", () => {
    const { getByTestId } = render(<TextArea disabled={false} />);
    expect(getByTestId("arke-textarea")).toBeInTheDocument();
  });
});
