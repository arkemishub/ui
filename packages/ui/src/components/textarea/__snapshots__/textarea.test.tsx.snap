// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="block text-sm mb-2 textarea__label"
  />
  <textarea
    class="z-10 w-full px-2 flex items-center min-h-[40px] min-w-[120px] border border-neutral rounded-theme bg-background text-background-contrast focus-within:outline-offset-0 focus-within:border focus-within:border-primary focus-within:outline focus-within:outline-2 focus-within:outline-primary/30 focus-within:disabled:cursor-not-allowed py-2 textarea"
    data-testid="arke-textarea"
  />
</DocumentFragment>
`;
